gen.mk

```diff
@@ -0,0 +1,26 @@
+# Helper for generating image repository files
+
+ifndef common_dir
+    common_dir = common
+endif
+
+generator = $(common_dir)/generate.sh
+DISTGEN_BIN ?= /usr/bin/dg
+MANIFEST_FILE ?= manifest.sh
+
+generation_env = \
+	DG=$(DISTGEN_BIN) \
+	MANIFEST_FILE=$(MANIFEST_FILE)
+
+.PHONY: gen
+gen: auto_targets.mk exec-gen-rules
+	rm auto_targets.mk
+
+
+auto_targets.mk: $(generator) $(MANIFEST_FILE)
```

Comment:
Why this one has `.mk` at the end? Please forgive my ignorance...

Reply:
Np, auto_targets.mk is generated makefile containing rules for all targets to be created.

Reply:
It is. `gen` depends on `auto_targets.mk` which in turn depends on `generator.sh` in which the actual file is generated.